rfcs/theming-color-scheme-strategy.md

```diff
@@ -0,0 +1,164 @@
+Theming Strategy in Orbit
+
+## Summary
+
+Orbit has been, in it's infancy, a monolitich piece of CSS, the only theming options were to modify the primary colors in order to adapt to each products in the Sharegate ecosystem(Apricot, Overcast and Desktop). With Orbit 2.0 we aim to support, among things, a dark theme, while opening the door for future theming possibilitiess(high Contrast, density themes). In this RFC we propose a new architecture for CSS Custom Properties and a way to organize them.
+
+## Motivation
+
+Being able to quickly add a design token, and semantic variables, without questionning where to do so, is a main goal of ours. This is why we came up with a strategy, and are moving many custom properties. In order to provide a consistent theme, that behaves in a predictable manne, we are introducing the concept of semantic colors. Semantic colors are a layer over design tokens representing a meaning, e.g. `border-danger` has a semantic meaning while `border-red-500` has none and could be used on many situations. This make our intefaces more cohesive and gives intention to our design choices. As Adobe Spectrum sums it up:
+
+> Too much color can create cognitive overload, affecting users’ ability to efficiently interact with products.
+
+### Design Tokens
+
+Design tokens are the values needed to construct and maintain our design system, from spacing, color, typograghy, etc. These values are reflected in the form of CSS Custom Properties(variables). These are prefixed with `o-ui` as well as a `global`. Design tokens are not affected by a theme, they are what one could call a constant.
+
+On top of design tokens lives *semantic variables*, these have an assigned meaning and expedite design decisions. One example of a semantic variable would be a `border-danger` variable, this variable would use the `beetle-500` color behind the scene. When in need of a border that represent a negative state one designer wouldn't need to reinvent the wheel and should use this abstraction : `border-danger`.  These are prefixed with `o-ui-alias`.
+
+Here is an example on how semantic variables work at a core level: 
+
+`--o-ui-alias-text-primary` is a semantic variable who uses `--o-ui-global-marine-900` when in a light theme and `--o-ui-global-marine-500` when in a dark theme.
+
+### Supported Themes
+
+Orbit supports two themes, light and dark as well as two color schemes, Apricot and Desktop. Every semantic variables need to support both themes, light and dark. Some variables can be affected by a color scheme, e.g. a primary background is different in Apricot than in Desktop. Color schemes colors can also be affected by a theme.
```

Comment:
I think this paragraph could be written in a way where we don't enumerate all our current themes and color schemes. This would make this section just as good, but would make it be accurate longer. We wouldn't need to update it if we added a theme for instance